src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2244,46 +2243,47 @@ void ApplyPreserveInfo (TypeDefinition type)
 		{
 			ApplyPreserveMethods (type);
 
-			if (!Annotations.TryGetPreserve (type, out TypePreserve preserve))
-				return;
-
-			var di = new DependencyInfo (DependencyKind.TypePreserve, type);
+			if (Annotations.TryGetPreserve (type, out TypePreserve preserve)) {
+				var di = new DependencyInfo (DependencyKind.TypePreserve, type);
 
-			switch (preserve) {
-			case TypePreserve.All:
-				Annotations.TryGetPreserveAccessibility (type, out preserve);
 				switch (preserve) {
-				case TypePreserve.AccessibilityVisible:
+				case TypePreserve.All:
+					MarkFields (type, true, di);
+					MarkMethods (type, di, type);
```

Comment:
This is a change in behavior - before if it was TypePreserve.All and AllVisible we would only call `MarkMethodsIf (..., IsMethodVisible, ...)` and so on.
Now we call `MarkMethods` and then also `MarkMethodsIf`. 

Also the `MarkMethodsIf` later on will not have any effect really (since all methods are already marked).

Reply:
> if it was TypePreserve.All and AllVisible we would only call MarkMethodsIf (..., IsMethodVisible, ...) and so on.

yeah, that's one of the bugs this is fixing

> Also the MarkMethodsIf later on will not have any effect really (

Fixed that